doc/api/cli.md

```diff
@@ -995,6 +995,18 @@ added:
 
 Use this flag to enable [ShadowRealm][] support.
 
+### `--experimental-strip-input-types`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active development
+
+Enable experimental type-stripping for `--eval`.
+Implies [`--experimental-strip-types`][].
+For more information, see the [TypeScript type-stripping][] documentation.
```

Comment:
Ideally this could just be a new value for `--input-type`, so like `--input-type=strip-types`; could that work? And I guess it would rely on syntax detection to determine format, or it would just always assume ESM since TypeScript syntax seems to be always ESM?

Reply:
Yeah but then dont see how it can ever be enabled by default.

Reply:
You could take the same approach as syntax detection: try to run as regular `--eval`, if it throws based on an exception that might’ve been caused by TypeScript syntax, retry as `--input-type=strip-types`.

Reply:
Yes thats what Im experimenting